src/api.js

```diff
@@ -193,6 +195,8 @@ class API {
       demo: this.config.demo ? true : undefined,
       // 'provider' param in priority, skip 'providers' param if 'provider' is set
       providers: !provider && Array.isArray(providers) && providers.length ? providers.join(',') : undefined,
+      theme: ['light', 'dark'].includes(theme) ? theme : 'light',
+      providers_per_line: (providersPerLine && ['1', '2'].includes(providersPerLine.toString())) ? providersPerLine.toString() : '2',
```

Comment:
does it not work with number and we need to explicitly parse the number to string?

Reply:
I also wonder if we can just use `columns` instead of the long `providers_per_line`

Reply:
> does it not work with number and we need to explicitly parse the number to string?
No, `includes()` do not auto cast types, so type should be also matched.
E.g. this statement `['1', '2'].includes(1)` returns `false`.

Also I want to accept both string and number for `providersPerLine` passed value.

> I also wonder if we can just use columns instead of the long providers_per_line
Then it will not be clear what it is. Possible that we have view with few columns in some other templates in future and that `columns` will be misleading. Really I don't think that long name in url is a problem, I prefer to leave a friendly name.

Reply:
Also I thought about `provider_columns`, but for some reason preferred `providers_per_line`.